### PR TITLE
BAM-3362 add new section to publisher tag

### PR DESCRIPTION
**Pull request by PhilWinchester**
We want to try and encourage publishers to write their own tight integrations. Add a section to the docs that shows how they can do this and a sample snippet for inspiration. 

---

image

### CODE REVIEW

File: docs/tagpublisher.rst [hunk 1]
```diff
@@ -75,5 +75,43 @@ Toggle LinkMate
 
     http://amazon.com.example/BF93JSD34/ref=ods?#locklink
 
+Utilizing Narrativ Data
+-----------------------
+
+As discussed above the main benefit of the Narrativ JsTag is to run auctions on page load. After an auction completes the Narrativ tag will write the output of the auction to the `data-bamx-auction` attribute. If you would like to update the text on or around the link after an auction runs you can use the data we store there to pull accurate product information. This product is good for publishers who’s commerce buttons or links mention the merchant's name and price (“$5 at Nordstrom”). This option would update the text on the page to match the auction winner.
```
Comment (myfashionhub) [inline docs/tagpublisher.rst]: I would drop "As discussed above".
`After an auction request completes, the Narrativ tag will write the output of the auction to the `data-bamx-auction` attribute.`
`The product data is good`
`If you would like to update text after an auction runs, you can use`
`who’s` => `whose`
`This option would update the text on the page to match the auction winner.` What's "this option"

File: docs/tagpublisher.rst [hunk 2]
```diff
@@ -75,5 +75,43 @@ Toggle LinkMate
...
+  :linenos:
+  :emphasize-lines: 11
+
+  const anchorNodes = [...document.querySelectorAll('a.monetized-links')];
+  const config = {attributes: true};
+
+  for (let i = 0; i < anchorNodes.length; i++) {
+  	let anchor = anchorNodes[i];
+
+  	const logFunction = (mutationList, observer) => {
```
Comment (myfashionhub) [inline docs/tagpublisher.rst]: define this outside the for loop

File: docs/tagpublisher.rst [hunk 3]
```diff
@@ -75,5 +75,43 @@ Toggle LinkMate
...
+  		for (let j = 0; j < mutationList.length; j++) {
+  			const mutation = mutationList[j];
+
+        if (mutation.type === 'attributes' && mutation.attributeName === 'data-bamx-auction') {
+            console.log('Narrativ Auction has finished. Update display values now');
+            console.log(anchor.getAttribute('data-bamx-auction'));
+        }
```
Comment (myfashionhub) [inline docs/tagpublisher.rst]: indentation looks funky here

File: docs/tagpublisher.rst [hunk 4]
```diff
@@ -75,5 +75,43 @@ Toggle LinkMate
...
+  		}
+  	};
+
+  	let observer = new MutationObserver(logFunction);
```
Comment (myfashionhub) [inline docs/tagpublisher.rst]: const

File: docs/tagpublisher.rst (hunk 2 above)
Comment (PhilWinchester) [reply docs/tagpublisher.rst]: In the example we need a reference to the current anchor tag. I think keeping it this way is a bit clearer, and the whole point of this is to just be an example not the actual code to be used.

File: docs/tagpublisher.rst (hunk 3 above)
Comment (PhilWinchester) [reply docs/tagpublisher.rst]: good catch! Bad copy paste

**Comment by kaitlynkrull** [comment]
it looks so good to me

File: docs/tagpublisher.rst [hunk 5]
```diff
@@ -75,5 +75,44 @@ Toggle LinkMate
 
     http://amazon.com.example/BF93JSD34/ref=ods?#locklink
 
+Utilizing Narrativ Data
+-----------------------
+
+As discussed above, the main benefit of the Narrativ JsTag is to run auctions on page load. This product works well for publishers who’s commerce buttons or links mention the merchant's name and price (“$5 at Nordstrom”). If you would like to dynamically display the auction winner and product price of a link, you can use the example to update your page.
```
Comment (zashuuu) [inline docs/tagpublisher.rst]:
```suggestion
As discussed above, the main benefit of the Narrativ JsTag is to find links on the page and run their auctions on page load. For publishers whose commerce buttons or links mention the merchant's name and product price (“$5 at Nordstrom”), this feature will enable you to replace those values dynamically. 
```

File: docs/tagpublisher.rst [hunk 6]
```diff
@@ -75,5 +75,44 @@ Toggle LinkMate
 
     http://amazon.com.example/BF93JSD34/ref=ods?#locklink
 
+Utilizing Narrativ Data
```
Comment (zashuuu) [inline docs/tagpublisher.rst]:
```suggestion
Updating Article Content with Auction Results
```

**Pull request merged**